sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveMetastoreCatalog.scala

```diff
@@ -195,17 +207,20 @@ private[hive] class HiveMetastoreCatalog(sparkSession: SparkSession) extends Log
           fileFormatClass,
           None)
         val logicalRelation = cached.getOrElse {
+          val fileIndex = new InMemoryFileIndex(sparkSession, Seq(rootPath), options, None)
```

Comment:
This `InMemoryFileIndex` will list leaf files even `inferenceMode` is `NEVER_INFER`. Isn't it unnecessary?